src/buffer_object.rs

```diff
@@ -568,9 +569,9 @@ impl<T: 'static> BufferObject<T> {
     }
 }
 
-impl<T: 'static> AsRawFd for BufferObject<T> {
-    fn as_raw_fd(&self) -> RawFd {
-        unsafe { ffi::gbm_bo_get_fd(*self.ffi) }
+impl<T: 'static> AsFd for BufferObject<T> {
+    fn as_fd(&self) -> BorrowedFd {
+        unsafe { BorrowedFd::borrow_raw(ffi::gbm_bo_get_fd(*self.ffi)) }
```

Comment:
This implementation is flawed. `gbm_bo_get_fd` returns a new owned file descriptor, that needs to be closed. Casting to a `BorrowedFd` will leak the file descriptor. If we want to drop `AsRawFd`, we need to drop this entirely.